internal/controller/fluxinstance_controller.go

```diff
@@ -622,3 +640,34 @@ func (r *FluxInstanceReconciler) recordMetrics(obj *fluxcdv1.FluxInstance) error
 	reporter.RecordMetrics(unstructured.Unstructured{Object: rawMap})
 	return nil
 }
+
+func (r *FluxInstanceReconciler) notify(ctx context.Context, obj *fluxcdv1.FluxInstance, reason, eventType, msg string) {
+	notificationAddress := ""
+	if os.Getenv("NOTIFICATIONS_DISABLED") == "" && builder.ContainElementString(obj.GetComponents(), builder.MakeDefaultOptions().NotificationController) {
```

Comment:
I'd prefer avoiding an env var here and introduce this as a boolean in the `FluxInstanceReconciler` struct instead.

Reply:
Notifications should work for ResourceGroups and any other new APIs we'll add in the future, so I want to avoid hard coupling it to the FluxInstance. The env var is more of a kill switch for when you don't want the operator to send any requests to the notification-controller HTTP endpoint.